plugins/in_elasticsearch/in_elasticsearch_bulk_prot.c

```diff
@@ -801,13 +806,28 @@ int in_elasticsearch_bulk_prot_handle(struct flb_in_elasticsearch *ctx,
         request->method != MK_METHOD_GET &&
         request->method != MK_METHOD_HEAD &&
         request->method != MK_METHOD_PUT) {
+
+        if (bulk_statuses) {
+            flb_sds_destroy(bulk_statuses);
+        }
+        if (bulk_response) {
+            flb_sds_destroy(bulk_response);
+        }
+
         flb_sds_destroy(tag);
         mk_mem_free(uri);
 
         send_response(conn, 400, "error: invalid HTTP method\n");
         return -1;
     }
 
+    if (bulk_statuses == NULL) {
+        bulk_statuses = flb_sds_create_size(ctx->buffer_max_size);
+    }
+    if (bulk_response == NULL) {
+        bulk_response = flb_sds_create_size(ctx->buffer_max_size);
+    }
+
```

Comment:
They won't be needless for non POST case. Elasticsearch BULK API requests to return back their API statuses but other endpoint shouldn't use them. 

ref: https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-bulk.html#docs-bulk-api-example

The original OpenSearch And Elasticsearch responses are : 

```json
{
   "took": 30,
   "errors": false,
   "items": [
      {
         "index": {
            "_index": "test",
            "_id": "1",
            "_version": 1,
            "result": "created",
            "_shards": {
               "total": 2,
               "successful": 1,
               "failed": 0
            },
            "status": 201,
            "_seq_no" : 0,
            "_primary_term": 1
         }
      },
      {
         "delete": {
            "_index": "test",
            "_id": "2",
            "_version": 1,
            "result": "not_found",
            "_shards": {
               "total": 2,
               "successful": 1,
               "failed": 0
            },
            "status": 404,
            "_seq_no" : 1,
            "_primary_term" : 2
         }
      },
      {
         "create": {
            "_index": "test",
            "_id": "3",
            "_version": 1,
            "result": "created",
            "_shards": {
               "total": 2,
               "successful": 1,
               "failed": 0
            },
            "status": 201,
            "_seq_no" : 2,
            "_primary_term" : 3
         }
      },
      {
         "update": {
            "_index": "test",
            "_id": "1",
            "_version": 2,
            "result": "updated",
            "_shards": {
                "total": 2,
                "successful": 1,
                "failed": 0
            },
            "status": 200,
            "_seq_no" : 3,
            "_primary_term" : 4
         }
      }
   ]
}
```

However, the minimum requirements on fluent-bit's plugin are:

```json
{
   "errors": false,
   "items": [
      {
         "index": {
            "result": "created",
            "status": 201
         }
      },
      {
         "delete": {
            "result": "not_found",
            "status": 404
         }
      },
      {
         "create": {
            "result": "created",
            "status": 201
         }
      },
      {
         "update": {
            "result": "updated",
            "status": 200
         }
      }
   ]
}
```

Reply:
Thank you for reviewing.
I worried about touching `bulk_statuses` and `bulk_response` in non POST cases.
So, I committed it since I missed "early returns" in non POST cases.

I agree these allocations are needless.